substrate/frame/revive/src/lib.rs

```diff
@@ -1212,7 +1390,20 @@ sp_api::decl_runtime_apis! {
 			code: Code,
 			data: Vec<u8>,
 			salt: Option<[u8; 32]>,
-		) -> ContractInstantiateResult<Balance, EventRecord>;
+		) -> ContractResult<InstantiateReturnValue, Balance, EventRecord>;
+
+
+		/// Perform an Ethereum call.
+		///
+		/// See [`crate::Pallet::bare_eth_transact`]
+		fn eth_transact(
+			origin: H160,
+			dest: Option<H160>,
+			value: Balance,
+			input: Vec<u8>,
+			gas_limit: Option<Weight>,
+			storage_deposit_limit: Option<Balance>,
+		) -> EthContractResult<Balance>;
```

Comment:
Why do we need that? Can't the proxy dispatch into the existing runtime APIs?

Reply:
this let us do the branching logic here between the two type of transaction, plus we compute the exact fee here,